src/main/java/io/prestosql/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -1079,9 +1133,18 @@ public Object getObject(int field)
     @Override
     public boolean isNull(int field)
     {
-        checkState(!closed, "cursor is closed");
-        checkArgument(field < columnHandles.size(), "Invalid field index");
-        // TileDB does not allow null values
+        short[] validityByteMap;
+        boolean nullable = fieldNullables[field];
+        String name = fieldNames[field];
+        if (nullable) {
+            try {
+                validityByteMap = query.getValidityByteMap(name);
+                return validityByteMap[cursorPosition] == 0;
+            }
+            catch (TileDBError tileDBError) {
+                /*do nothing*/
```

Comment:
Let's log an error here and point from which part of the code it comes (`LOG.error(...)`)